src/swarm/Swarm.ts

```diff
@@ -137,23 +137,44 @@ interface FeedTemplate {
     protocolVersion: number;
 }
 
-export const downloadUserFeedTemplate = async (identity: PublicIdentity): Promise<FeedTemplate> => {
-    const url = DefaultGateway + `/bzz-feed:/?user=${identity.address}&meta=1`;
-    Debug.log('downloadUserFeedTemplate: ', url);
-    const response = await safeFetch(url);
-    const feedUpdateResponse = await response.json() as FeedTemplate;
+const calculateFeedAddressQueryString = (address: FeedAddress): string => {
+    return `user=${address.user}` + (address.topic === '' ? '' : `&topic=${address.topic}`);
+};
+
+const downloadUserFeedTemplate = async (swarmGateway: string, address: FeedAddress): Promise<FeedTemplate> => {
+    const addressPart = calculateFeedAddressQueryString(address);
+    const response = await downloadFeed(swarmGateway, `bzz-feed:/?${addressPart}&meta=1`);
```

Comment:
as mentioned before, we interpolate / concat lot of swarm related urls, these should be handled at one place

Reply:
I agree, but I didn't want to spend too much time on the Swarm library, because I am already looking for a replacement (e.g. Erebos), because maintaining this is just too much effort.